symengine/subs.h

```diff
@@ -407,6 +410,50 @@ inline RCP<const Basic> ssubs(const RCP<const Basic> &x,
     return s.apply(x);
 }
 
+class SubsVisitor : public BaseVisitor<SubsVisitor, XReplaceVisitor>
+{
+public:
+    using XReplaceVisitor::bvisit;
+
+    SubsVisitor(const map_basic_basic &xreplace_dict_)
+        : BaseVisitor<SubsVisitor, XReplaceVisitor>(xreplace_dict_)
+    {
+    }
+
+    void bvisit(const Pow &x)
```

Comment:
on second thought, `Add` and `Mul` are not performing any such simplifications. So, It might be better to keep it as it is.